jax/interpreters/partial_eval.py

```diff
@@ -227,23 +226,24 @@ def process_map(self, map_primitive, f: lu.WrappedFun, tracers, params):
       tracers = map(self.instantiate_const_abstracted, tracers)
     else:
       name = wrap_name(name, 'pe')
-
     params = dict(params, name=name)
-    in_pvs, in_consts = unzip2([t.pval for t in tracers])
-    reduced_pvs = [None if pv is None else
-                   core.mapped_aval(params['axis_size'], pv) if m else pv
-                   for pv, m in zip(in_pvs, params['mapped_invars'])]
-    fun, aux = partial_eval(f, self, reduced_pvs)
-    out_flat = map_primitive.bind(fun, *in_consts, **params)
-    out_pvs_reduced, jaxpr, env = aux()
-    out_pv_consts, consts = split_list(out_flat, [len(out_flat)-len(jaxpr.constvars)])
-    out_pvs = [None if pv is None else core.unmapped_aval(params['axis_size'], pv)
-               for pv in out_pvs_reduced]
+
+    @curry
+    def modify_aval(mod, args):
+      pval, is_mapped = args
+      if pval.is_known() or not is_mapped:
+        return pval
+      return PartialVal((mod(params['axis_size'], pval[0]), pval[1]))
```

Comment:
Might be best not to shadow the builtin name `mod` here. How about just `modify`?

Reply:
Huh, didn't know it's a builtin!

Reply:
Seems like it's not? At least not in Python 3.7